Modules/Scaler/DialsScaler.py

```diff
@@ -1269,3 +1284,20 @@ def decide_correct_lattice_using_refiner(possible_lattices, refiner):
         logger.debug("No solution found: assuming lattice from refiner")
 
     return correct_lattice, rerun_symmetry, need_to_return
+
+
+def convert_mtz_to_sca(mtz_filename, merged=False):
+    """Convert an mtz files to .sca format and write."""
+    sca_filename = mtz_filename.replace("mtz", "sca")
+    m = mtz.object(mtz_filename)
+    cols = m.as_miller_arrays(merge_equivalents=False, anomalous=False)
+    i_obs = None
+    for ma in cols:
+        if ma.info().labels == ["I", "SIGI"] and not merged:
+            i_obs = ma
+        elif ma.info().labels == ["IMEAN", "SIGIMEAN"] and merged:
+            i_obs = ma
+    if merged:
+        merge_scalepack_write(miller_array=i_obs, file_name=sca_filename)
+    else:
+        no_merge_original_index.writer(i_obs, file_name=sca_filename)
```

Comment:
maybe clearer, at least infinitesimally faster due to short-circuit evaluation, added `FileHandler` call, and maybe a `raise`:
```python
for ma in cols:
    if merged and ma.info().labels == ["IMEAN", "SIGIMEAN"]:
         merge_scalepack_write(miller_array=ma, file_name=sca_filename)
         break
    elif not merged and ma.info().labels == ["I", "SIGI"]:
         no_merge_original_index.writer(ma, file_name=sca_filename)
         break
else:
    raise OhNoException("No relevant labels found")
FileHandler.record_data_file(sca_filename)
```

Reply:
Thanks, today I learned about for/else loops in python :)